### PR TITLE
Refactor/tab

### PR DESCRIPTION
**Pull request by kherP**
- added keyboard support based on [W3C guidelines](https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html)
- introduced new properties:
  - autoSelectOnFocus: set focused element to active by auto activation
  - onTabDelete: to allow user to do some actions on delete button press
- refactored playground component to use tab component

### CODE REVIEW

File: lib/src/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -1,22 +1,62 @@
...
+        }
+    };
     const onClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
-        event.currentTarget.href.endsWith("#") && event.preventDefault();
-        onTabChange && onTabChange(parseInt(event.currentTarget.dataset.indexNumber));
+        navigateToTab(event.currentTarget, event);
+    };
+
+    const onKeyDown = (event: React.KeyboardEvent<HTMLUListElement>) => {
+        const items: HTMLAnchorElement[] = Array.from(tabsRef.current.querySelectorAll(".nav-tabs > .rc.nav-item > .nav-link"));
```
Comment (eweseong) [inline lib/src/Tabs/Tabs.tsx:34]: let's move this closer to its usage in the `ArrowX` case to reduce HTML queries

File: lib/src/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -1,22 +1,62 @@
...
+        switch (event.key) {
+            case "Enter":
+            case " ":
+                navigateToTab(event.target as HTMLAnchorElement, event);
+                break;
+            case "ArrowLeft":
+            case "ArrowRight":
+                const direction: number = event.key === "ArrowLeft" ? -1 : 1;
+                const nextFocusedIndex: number = focusedIndex + direction;
```
Comment (eweseong) [inline lib/src/Tabs/Tabs.tsx:43]: we can modify the algorithm slightly to support infinite moving (move to last when on first; move to first when on last)

```
const nextFocusedIndex: number = (focusedIndex + items.length + direction) % items.length;
```

File: lib/src/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -1,22 +1,62 @@
...
+    };
     const onClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
-        event.currentTarget.href.endsWith("#") && event.preventDefault();
-        onTabChange && onTabChange(parseInt(event.currentTarget.dataset.indexNumber));
+        navigateToTab(event.currentTarget, event);
+    };
+
+    const onKeyDown = (event: React.KeyboardEvent<HTMLUListElement>) => {
+        const items: HTMLAnchorElement[] = Array.from(tabsRef.current.querySelectorAll(".nav-tabs > .rc.nav-item > .nav-link"));
+        switch (event.key) {
```
Comment (eweseong) [inline lib/src/Tabs/Tabs.tsx:35]: maybe we can add support for `Home` and `End` button navigation here as well?

**Review by mario-subo — APPROVED** [review]

**Review by eweseong — APPROVED** [review]

**Pull request merged**